File: leetcode/solutions/g51/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {

        int n = nums.length;
        
        int l = 0, r = n - 1, c = 0;
        while (c <= r){
            if (nums[c] == 2){
                swap(nums, c, r--);
                
            }
            else if (nums[c] == 1){
                
                c++;
            }
            else {
                swap(nums, c, l);
                c++;
                l++;
                
            }

        }

        
    }
    private void swap(int [] nums, int c, int i){
        int temp = nums[c];
        nums[c] = nums[i];
        nums[i] = temp;
    }
}

File: a2sv/g51/codebase/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # using bucct sort 
        s = {}
        for num in nums:
            s[num] = s.get(num, 0) +1
        i = 0
        while i < (s.get(0, 0)):
            nums[i] = 0
            i+=1
        while i < s.get(0, 0) + s.get(1,0):
            nums[i] = 1
            i+=1
        while i < s.get(0, 0) + s.get(1, 0) + s.get(2, 0):
            nums[i] = 2
            i+=1
